plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MappedNamespaceContext.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.util;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+
+public class MappedNamespaceContext implements NamespaceContext {
+
+  private final Map<String, String> prefixMapping = new HashMap<>();
+
+  public MappedNamespaceContext() {
+    prefixMapping.put("xml", XMLConstants.XML_NS_URI);
+  }
+
+  public MappedNamespaceContext(String prefix, String namespaceUri) {
+    this();
+    addMappingInternal(prefix, namespaceUri);
+  }
+
+  public MappedNamespaceContext addMapping(String prefix, String namespaceUri) {
```

Comment:
addMapping --> declareNamespace for consistency with terminology in the namespaces specification.